Analyticsv3Sample/AnalyticsSampleProject/StandardGoogleAuthSample.cs

```diff
@@ -0,0 +1,239 @@
+                                             AnalyticsService.Scope.AnalyticsEdit,                   // Edit Google Analytics management entities
+                                             AnalyticsService.Scope.AnalyticsManageUsers,            // Manage Google Analytics Account users by email address
+                                             AnalyticsService.Scope.AnalyticsManageUsersReadonly,    // View Google Analytics user permissions
+                                             AnalyticsService.Scope.AnalyticsProvision,              // Create a new Google Analytics account along with its default property and view
+                                             AnalyticsService.Scope.AnalyticsReadonly};              // View your Google Analytics data
+
+            try
+            {
+
+                string credPath = System.Environment.GetFolderPath(System.Environment.SpecialFolder.Personal);
+                credPath = Path.Combine(credPath, ".credentials/Analytics");
+
+                // here is where we Request the user to give us access, or use the Refresh Token that was previously stored in %AppData%
+                UserCredential credential = GoogleWebAuthorizationBroker.AuthorizeAsync(new ClientSecrets { ClientId = clientId, ClientSecret = clientSecret }
+                                                                                             , scopes
+                                                                                             , userName
+                                                                                             , CancellationToken.None
+                                                                                             , new FileDataStore(credPath, true)).Result;
+
+                var service = new AnalyticsService(new BaseClientService.Initializer()
+                {
+                    HttpClientInitializer = credential,
+                    ApplicationName = "Analytics Authentication Sample",
+                });
+                return service;
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.InnerException);
+                throw ex;
+            }
+
+        }
+
+        /// <summary>
+        /// Authenticating to Google using a Service account
+        /// Documentation: https://developers.google.com/accounts/docs/OAuth2#serviceaccount
+        /// </summary>
+        /// <param name="serviceAccountEmail">From Google Developer console https://console.developers.google.com</param>
+        /// <param name="keyFilePath">Location of the Service account key file downloaded from Google Developer console https://console.developers.google.com</param>
+        /// <returns>a valid AnalyticsService</returns>
+        public static AnalyticsService AuthenticateServiceAccount(string serviceAccountEmail, string keyFilePath)
```

Comment:
I don't think that all services support ServiceAccount, how do you know if you need to generate it or not. Do you just generate it for every service.
If so, add some context that says that the service itself might not support service account...

Makes sense?